realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -154,6 +154,50 @@ TableOrView getTable() {
     }
 
     /**
+     * Check if the underlying {@link Table} exists.
+     *
+     * @return {@code true} if the backing Table is valid, {@code false} otherwise.
+     */
+    private boolean isTablePresent() {
```

Comment:
I would somehow cache the result of this. Right now you are introducing an extra JNI call to every method in this class.